packages/backend/src/peripherals/optimism/multicall/OptimismMulticall.ts

```diff
@@ -0,0 +1,107 @@
+import { ChainId, EthereumAddress } from '@l2beat/shared-pure'
+
+import { EthereumClient } from '../../ethereum/EthereumClient'
+import {
+  MulticallEncoder,
+  MulticallRequest,
+  MulticallResponse,
+} from './interfaces'
+import { optimismMulticallEncoder } from './MulticallEncoder'
+
+// TODO: Investigate if max batch size can be tweaked
+/**
+ * @see https://optimistic.etherscan.io/address/0xe295ad71242373c37c5fda7b57f26f9ea1088afe
+ */
+export const OPTIMISM_MULTICALL_BATCH_SIZE = 150
+export const OPTIMISM_MULTICALL_BLOCK = 0
+export const OPTIMISM_MULTICALL_ADDRESS = EthereumAddress(
+  '0xE295aD71242373C37C5FdA7B57F26f9eA1088AFe',
+)
+
+export class OptimismMulticallClient {
```

Comment:
We can probably make the existing MulticallClient take the encoder/decoder as configuration